server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -170,6 +172,9 @@ private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
           log.debug("Candidate was still in use: {}", relativePath);
       }
     }
+
+    // close underlying scanner
+    refStream.close();
```

Comment:
Does this actually close the underlying scanner? Or just the stream?

Reply:
That is a good question. I was thinking it closes both but maybe I can write a test to see. The scanner was being closed by the `try-with-resources` on the `TabletsMetadata` but that class is the one that is `AutoCloseable`. If we want to be able to close the underlying scanner (which I think would be good) then we probably have to refactor the GC code.